rohmu/compressor.py

```diff
@@ -54,17 +57,17 @@ def __init__(self, src_fp, algorithm, level=0):
         else:
             raise InvalidConfigurationError("invalid compression algorithm: {!r}".format(algorithm))
 
-    def _process_chunk(self, data):
+    def _process_chunk(self, data: bytes) -> bytes:
         return self._compressor.compress(data)
 
-    def _finalize(self):
+    def _finalize(self) -> Any:
         return self._compressor.flush()
 
 
-def DecompressionFile(src_fp, algorithm):
+def DecompressionFile(src_fp: io.BufferedIOBase, algorithm: str) -> io.BufferedIOBase:
```

Comment:
```suggestion
def DecompressionFile(src_fp: HasRead, algorithm: str) -> io.BufferedIOBase:
```

Reply:
Yeah, changes this actually requires more changes downstream to FileWrap & all the various subclasses. `read` is not enough for them.

I believe we use at least: `read`, `write`, `flush`, `close` and `fileno`, possibly other methods in that hierarchy so I guess we need an other protocol for that other than `HasRead`.

I'm going to try and see what we end up with.